### PR TITLE
WIP: Better support for override redirect

### PR DESCRIPTION
**Pull request by Joshua-Ashton**
Super super early -- but WineCFG and some other stuff is usable now. :)

### CODE REVIEW

File: src/steamcompmgr.cpp [hunk 1]
```diff
@@ -839,7 +839,8 @@ void MouseCursor::constrainPosition()
 	int rootX, rootY;
 	queryGlobalPosition(rootX, rootY);
 
-	if (rootX >= window->a.width || rootY >= window->a.height) {
+	if (rootX - window->a.x >= window->a.width || rootY - window->a.y >= window->a.height ||
+		rootX - window->a.x < 0 || rootY - window->a.y < 0 ) {
 		warp(window->a.width / 2, window->a.height / 2);
```
Comment (Joshua-Ashton) [inline src/steamcompmgr.cpp:847]: (for my ref) probably want to do this offsetted by window->a.x if we do offseted override redirect windows

**Comment by emersion** [comment]
Apparently we really like duplicating work… https://github.com/Plagman/gamescope/pull/305

**Comment by Joshua-Ashton** [comment]
buh :frog:

**Comment by Joshua-Ashton** [comment]
With this branch now, the FONV settings are fully usable. Still some hacks to remove though.

**Comment by Joshua-Ashton** [comment]
WineCFG is also fully usable. Any other games we know that have old fashioned menus like this?

**Comment by emersion** [comment]
The launcher for Jedi Fallen Order (with the registration form to select your date of birth) was not working iirc, maybe try that? Titanfall 2 also had that Origin launcher I think.

**Pull request closed without merging**